Automattic/sensei
# Importer: Link prerequisites for courses and lessons

### Changes proposed in this Pull Request

* Adds post-processing ability to the file processor. Post-processing tasks happen in batches after all lines in the task file have been processed.
* Adds shared prerequisite post-processing tasks to match a prerequisite based on reference and associate them with the lesson or course in the appropriate post meta.
* Required Refactor: Pass line number and the task (instead of the job) to the models. This allowed us to log the line number and put shared logic for courses and lessons in the task level.
* Required Refactor: Move the `translate_import_id` method from the model to job class.
* Fix: Passing the "real" line number down so that the zero-index line number isn't logged for users.
* Known issue: These post-process tasks don't affect the progress of the import. Progress will hang while we handle these. It'd be a bit of a pain to fix this, but be we could probably figure out a way.

### Testing instructions

* Using Donna's test files, run the WP CLI command (filling in the parameters):
```
wp sensei-import --lessons=import/donna-lessons.csv --courses=import/donna-courses.csv --questions=import/donna-questions.csv --user=jake
```
* Verify prerequisites are stored when appropriate.
* Verify logs are added for bad prerequisites.
